src/screens/nearest.tsx

```diff
@@ -0,0 +1,203 @@
+import { LocationObject } from 'expo-location'
+import { useState, useEffect } from 'react'
+import { FlatList, StyleSheet, View, Text, Platform } from 'react-native'
+import { ActivityIndicator, MD2Colors } from 'react-native-paper'
+import * as Location from 'expo-location'
+import { dftCallToJSON, DfTData } from '../clients/dftAPI'
+import {
+  tfgmCall,
+  TfGMData,
+  IncomingTram,
+  pidDataToStopData,
+} from '../clients/tfgmAPI'
+import TramDetailsBox from '../components/tramDetailsBox'
+import { config } from '../config'
+
+type Coordinates = {
+  latitude: number
+  longitude: number
+}
+
+function calculateDistance(locationA: Coordinates, locationB: Coordinates) {
+  // Euclidean for now
+  return Math.sqrt(
+    Math.pow(locationA.latitude - locationB.latitude, 2) +
+      Math.pow(locationA.longitude - locationB.longitude, 2)
+  )
+}
+
+export default function NearestStopScreen() {
+  console.log('Rendering.....')
+  const [nearestStop, setNearestStop] = useState<{
+    name: string
+    atcoCode: string
+  }>()
+  const [incomingTrams, setIncomingTrams] = useState<IncomingTram[]>([])
+  const [stopsObtained, setStopsObtained] = useState<
+    { label: string; value: string }[]
+  >([])
+  const [location, setLocation] = useState<LocationObject>()
+  const [errorMsg, setErrorMsg] = useState<string>()
+
+  useEffect(() => {
+    const effectFunc = async () => {
+      const { status } = await Location.requestForegroundPermissionsAsync()
+      if (status !== 'granted') {
+        setErrorMsg('Permission to access location was denied')
+        return
+      }
+
+      const location = await Location.getCurrentPositionAsync({})
+      setLocation(location)
+      console.log('Got Location')
+    }
+    void effectFunc()
+  }, [])
+
+  useEffect(() => void getAllStops(), [getAllStops])
+  console.log('About to getNearestStop')
+  useEffect(
+    () => void getNearestStop(),
+    [location, stopsObtained, getNearestStop]
+  )
+  console.log('About to showTrams')
+  useEffect(() => void showTrams(), [nearestStop, showTrams])
```

Comment:
I don't think these logs are doing what you're expecting them to do. You should put them inside the `useEffect` lambda if you want to see when the `useEffect` re-runs.